### PR TITLE
deps: update google-cloud-bigtable dependencies

### PR DESCRIPTION
**Pull request by mutianf**
Copy of #3912  and fixed the tests

### CODE REVIEW

**Review by igorbernstein2 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -56,8 +56,8 @@ limitations under the License.
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
     <!-- core dependency versions -->
-    <bigtable.version>2.17.1</bigtable.version>
-    <google-cloud-bigtable-emulator.version>0.154.1</google-cloud-bigtable-emulator.version>
+    <bigtable.version>2.18.0</bigtable.version>
```
Comment (kolea2) [inline pom.xml:59]: latest is now 2.18.1 and 0.155.1 :)

File: bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestBigtableHBaseVeneerSettings.java [hunk 2]
```diff
@@ -285,15 +285,12 @@ public void testWithNullCredentials() throws IOException {
     configuration.setBoolean(BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY, true);
 
     BigtableDataSettings dataSettings =
-        ((BigtableHBaseVeneerSettings) BigtableHBaseVeneerSettings.create(configuration))
-            .getDataSettings();
-    assertTrue(
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestBigtableHBaseVeneerSettings.java:290]: was it intentional to delete this assertion?

Comment (mutianf) [reply bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestBigtableHBaseVeneerSettings.java:290]: Credential will be an instance of FixedCredentialProvider so the test will fail. The other line checks for  .getCredentials() which should be null and basically means that we are using a NoCredentialsProvider

Comment (kolea2) [reply bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestBigtableHBaseVeneerSettings.java:290]: thanks!

**Pull request closed without merging**